### PR TITLE
Completed Assignment

### PR DESCRIPTION
**Pull request by emtec-devyani-chaudhari**
Completed Assignment 3, 6 and 7.

### CODE REVIEW

File: 25-08-2022/Assignment3-sales-db.sql [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+SELECT * FROM customers;
+
+-- Write a query that produces all rows from the Customers table for which the salesperson’s number is 1001
+SELECT cnum, cname, city, rating, snum FROM customers WHERE snum = 1001;
+
+-- Write a select command that produces the rating followed by the name of each customer in San Jose
+SELECT rating, cname FROM customers WHERE city = 'San Jose';
+
+-- Write a query that will produce the snum values of all salespeople from the Orders table (with the duplicate values suppressed)
+SELECT snum, onum FROM orders ORDER BY onum;
```
Comment (rishikesh-dhanawade-emtec) [inline 25-08-2022/Assignment3-sales-db.sql]: this query will give me duplicate values but duplicate needs to be suppressed?
Isn't the order table need to be joined with salespeople table?

File: 25-08-2022/Assignment3-sales-db.sql [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+-- Write a query that will give you all orders for more than Rs. 1,000
+SELECT COUNT(*) AS Orders FROM orders WHERE amt >= 1000;
```
Comment (rishikesh-dhanawade-emtec) [inline 25-08-2022/Assignment3-sales-db.sql]: why Count(*)? It simply need have column names

File: 25-08-2022/Assignment3-sales-db.sql [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+-- Write a query that will give you the names and cities of all salespeople in London with a commission above 0.10
+SELECT sname, city FROM salespeople WHERE city = 'london' AND comm > 0.10;
+
+-- Write a query on the Customers table whose output will exclude all customers with a rating <=100, unless they are located in Rome
+SELECT cnum, cname, city, rating, snum FROM customers WHERE rating <= 100 AND city = 'Rome';
```
Comment (rishikesh-dhanawade-emtec) [inline 25-08-2022/Assignment3-sales-db.sql]: It asked to exclude all customers with a rating <=100, check this query again

File: 25-08-2022/Assignment3-sales-db.sql [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+SELECT cnum, cname, city, rating, snum FROM customers WHERE NOT (rating <= 100 AND city = 'Rome');
```
Comment (rishikesh-dhanawade-emtec) [inline 25-08-2022/Assignment3-sales-db.sql]: It could have simply done using OR

File: 25-08-2022/Assignment3-sales-db.sql [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+-- What will be the output from the following query? 
+SELECT * FROM orders WHERE (odate = '1990-10-03' AND cnum > 2003);
```
Comment (rishikesh-dhanawade-emtec) [inline 25-08-2022/Assignment3-sales-db.sql]: where is the output of these 2 queries?

File: 25-08-2022/Assignment3-sales-db.sql [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    "odate": "1990-10-04T00:00:00",
+    "cnum": 2006,
+    "snum": 1001
+  }
+]
+
+-- What will be the output of the following query?
+SELECT * FROM orders WHERE (odate = '1990-10-03' OR snum > 1006);
+Select * from orders WHERE (odate = '1990-10-03' OR snum > 1006) AND amt >= 1500;
+Select * from orders WHERE NOT ((odate = '1990-10-03' OR snum > 1006) AND amt >= 1500)
```
Comment (rishikesh-dhanawade-emtec) [inline 25-08-2022/Assignment3-sales-db.sql]: what about snum > 1006 and amt >= 1500 these conditions?

File: 25-08-2022/Assignment3-sales-db.sql [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+    "onum": 3011,
+    "amt": 9891.88,
+    "odate": "1990-10-04T00:00:00",
+    "cnum": 2006,
+    "snum": 1001
+  }
+]
+
+-- Write a query that selects all orders except those with zeroes or NULLs in the amt field
+SELECT onum, amt, odate,cnum,snum FROM orders WHERE amt IS NOT NULL;
```
Comment (rishikesh-dhanawade-emtec) [inline 25-08-2022/Assignment3-sales-db.sql]: need condition to check zero in amt column

File: 25-08-2022/Assignment6.sql [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+USE sales_db;
+-- Write a query that counts the number of salespeople registering orders for each day. 
+--(If a salesperson has more than one order on a given day, he or she should be counted only once.)
+SELECT odate,COUNT(distinct snum) AS total FROM orders GROUP BY odate;
+-- distinct - it will return different values from table 
+
+-- Write a query on the Customers table that will find the highest rating in each city. Put the output in
+-- this form:
+-- For the city (city), the highest rating is : (rating).
+SELECT city, MAX(rating) AS 'highest rating is :' FROM customers GROUP BY city;
```
Comment (rishikesh-dhanawade-emtec) [inline 25-08-2022/Assignment6.sql:10]: check this syntax again, rating is not printed after highest rating is :

File: 25-08-2022/Assignment6.sql [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+-- Write a query that totals the orders for each day and places the results in descending order.
+SELECT COUNT(odate) AS 'Total Orders',odate FROM orders GROUP BY odate ORDER BY 'Total Orders';
```
Comment (rishikesh-dhanawade-emtec) [inline 25-08-2022/Assignment6.sql]: shouldn't be column name in Order By? and it should be descending order

File: 25-08-2022/Assignment6.sql [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+ 
+-- Write a query that selects the highest rating in each city.
+SELECT city, MAX(rating) AS 'highest rating' FROM customers GROUP BY city;
+
+-- Largest order taken by each salesperson with order value more than Rs.3000. 
+SELECT o.odate, o.snum, MAX(amt) AS 'Largest Order' FROM orders o WHERE o.amt > 3000 GROUP BY odate,snum;
+
+SELECT * FROM orders;
+-- Select each customer smallest order
+SELECT cnum, MIN(amt) AS 'Smallest Order' FROM orders GROUP BY cnum;
```
Comment (rishikesh-dhanawade-emtec) [inline 25-08-2022/Assignment6.sql:26]: check if you can use joins where data is returned for salespeople, order and customer, order

**Review by rishikesh-dhanawade-emtec — CHANGES_REQUESTED** [review]
check comments

File: 25-08-2022/Assignment6.sql (hunk 10 above)
Comment (emtec-devyani-chaudhari) [reply 25-08-2022/Assignment6.sql:26]: orders and customers - SELECT c.cnum, MIN(amt) AS 'Smallest Order' FROM orders JOIN customers c ON orders.cnum = c.cnum GROUP BY c.cnum;

orders and salespeople - SELECT s.snum, MIN(amt) AS 'Smallest Order' FROM orders JOIN salespeople s ON orders.snum = s.snum GROUP BY s.snum;

**Review by rishikesh-dhanawade-emtec — APPROVED** [review]

**Pull request merged**